From a8c1602c107c4b5ff0168cd06cfbf7b4c97ef924 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:55 -0400
Subject: [tools] perf/evlist: Add a debug print if event buffer mmap fails

Message-id: <1408462094-14194-247-git-send-email-jolsa@redhat.com>
Patchwork-id: 88213
O-Subject: [PATCH RHEL7.1 BZ1131394 246/325] perf evlist: Add a debug print if event buffer mmap fails
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 026359658aecd348bc5c4a136a26f204b169103b
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Nov 1 15:51:33 2013 +0200

description
===========
Add a debug print if mmap of the perf event ring buffer fails.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/evlist.c b/tools/perf/util/evlist.c
index 0582f67..1c173cc 100644
--- a/tools/perf/util/evlist.c
+++ b/tools/perf/util/evlist.c
@@ -607,6 +607,8 @@ static int __perf_evlist__mmap(struct perf_evlist *evlist,
  evlist->mmap[idx].base = mmap(NULL, evlist->mmap_len, prot,
           MAP_SHARED, fd, 0);
  if (evlist->mmap[idx].base == MAP_FAILED) {
+  pr_debug2("failed to mmap perf event ring buffer, error %d\n",
+     errno);
   evlist->mmap[idx].base = NULL;
   return -1;
  }
-- 
1.7.1